File: 2021/src/main/java/nel/marco/Day1.java
package nel.marco;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;
import java.util.stream.Collectors;

public class Day1 {


    public static List<Integer> readDayOneInput() throws IOException {

        String path = "D:\\coding repo\\AdventOfCode\\2021\\src\\main\\resources\\day1";

        return Files.readAllLines(Path.of(path))
                .stream()
                .map(Integer::valueOf)
                .collect(Collectors.toList());

    }


    public static String partOne() throws IOException {

        int increase = 0;

        List<Integer> list = readDayOneInput();
        for (int i = 1; i < list.size(); i++) {
            if (list.get(i - 1) < list.get(i)) {
                increase++;
            }
        }

        return increase + "";
    }

    public static String partTwo() throws IOException {

        int increase = 0;

        List<Integer> list = readDayOneInput();
        for (int i = 1; i < list.size() - 2; i++) {

            int previousSum = list.get(i - 1) + list.get(i) + list.get(i + 1);
            int sum = list.get(i) + list.get(i + 1) + list.get(i + 2);


            if (previousSum < sum) {
                increase++;
            }
        }

        return increase + "";
    }
}

File: 2022/src/main/kotlin/Day1.kt
import java.nio.file.Files
import java.nio.file.Path

fun readInput(exampleInput: Boolean = false): String {
    if (exampleInput) {
        return Files.readAllLines(Path.of("D:\\coding repo\\AdventOfCode\\2022\\src\\main\\resources\\day1_example"))
            .joinToString("\n")
    }

    return Files.readAllLines(Path.of("D:\\coding repo\\AdventOfCode\\2022\\src\\main\\resources\\day1"))
        .joinToString("\n")
}

fun main(args: Array<String>) {

    val input = readInput()
    val elves = input.split("\n\n")

    println(answerOne(elves))
    println(answerTwo(elves))
}

fun answerOne(elves: List<String>): String {
    val listOfCalories = elves.map { s ->
        s.split("\n")
            .filter { it != "" }
            .sumOf {
                Integer.parseInt(it)
            }
    }.sortedDescending()

    return listOfCalories[0].toString()
}

fun answerTwo(elves: List<String>): String {
    val listOfCalories = elves.map { s ->
        s.split("\n")
            .filter { it != "" }
            .sumOf {
                Integer.parseInt(it)
            }
    }.sortedDescending()

    return listOfCalories.subList(0, 3).sum().toString()
}